clap_complete/src/dynamic/completer.rs

```diff
@@ -255,20 +259,20 @@ fn complete_path(
             let path = entry.path();
             let mut suggestion = pathdiff::diff_paths(&path, current_dir).unwrap_or(path);
             suggestion.push(""); // Ensure trailing `/`
-            completions.push(suggestion.as_os_str().to_owned());
+            completions.push((suggestion.as_os_str().to_owned(), None));
         } else {
             let path = entry.path();
             if is_wanted(&path) {
                 let suggestion = pathdiff::diff_paths(&path, current_dir).unwrap_or(path);
-                completions.push(suggestion.as_os_str().to_owned());
+                completions.push((suggestion.as_os_str().to_owned(), None));
```

Comment:
Not a blocker: is there anything of interest to show for paths, like the file mode?

Reply:
maybe, or the help of the argument, the path belongs to

Reply:
I like that!

A briefer version would be the `value_name`.  I can go for either

Reply:
Except that will get fairly repetitive....

Reply:
yeah, and it would result in a lot more space taken up by that redundant information. I guess we could add the description only to the first path, but not sure how that would work ergonomically?

Reply:
@AndreasBackx says that zsh supports groups of values which would be a great way to handle ones where things would get repetitive.

I guess for others shells, we just show nothing.  The extra context is not likely to be a deal breaker